etest/build.py

```diff
@@ -78,25 +78,37 @@ def main(
     if verbose:
         click.echo(f"Current profile: {profile.profile}")
 
-    build(quiet, verbose, profile, path)
+    with qemu.qemu(profile.arch):
+        build(quiet, verbose, profile, path)
 
 
 def build(quiet: bool, verbose: bool, profile: Profile, path: Path) -> None:
     """Build the image."""
-    docker.image.build(
-        path=str(path),
-        buildargs={"PROFILE": profile.docker_profile},
-        tag=f"etest/stage1:{profile.profile}",
-    )
-
-    stage2 = docker.container.run(
-        image=f"etest/stage1:{profile.profile}",
-        command=commands[profile.libc],
-        privileged=True,
-        name=f"stage2-{profile.profile}",
-    )["Container"]
-
-    docker.container.commit(container=stage2, repository="alunduil/etest", tag=profile.profile)
-
-    docker.container.remove(container=stage2)
-    docker.image.remove(image=f"etest/stage1:{profile.profile}")
+    try:
+        docker.image.build(
+            path=str(path),
+            buildargs={"PROFILE": profile.docker_profile},
+            tag=f"etest/stage1:{profile.profile}",
+        )
+
+        stage2 = docker.container.run(
+            image=f"etest/stage1:{profile.profile}",
+            command=commands[profile.libc],
+            privileged=True,
+            name=f"stage2-{profile.profile}",
+        )["Container"]
+
+        docker.container.commit(container=stage2, repository="alunduil/etest", tag=profile.profile)
+    except Exception as e:
+        click.echo("Exception triggered. Aborting.")
+
+        raise (e)
+    finally:
+        try:
+            # There's no problem if this errors, so leave it be. However, this specific order ensures cleanup always happens.
```

Comment:
@alunduil let me know what you think of this part. It isnt very elegant.

Reply:
I cringe every time I see a try in a finally.  Usually it means that we've got some wonky error handling or actions going on.  I'll take a deeper dive when I do a full review though.

Reply:
it is pretty wonky

Reply:
Did my inline comments as part of the review provide ideas on how to simplify this?